queries.sql

```diff
@@ -9,3 +9,55 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = TRUE;
 SELECT * FROM animals WHERE name <> 'Gabumon';
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+ /*----------------------------------*/
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals; 
+ROLLBACK;
+
+SELECT * FROM animals; --After ROLLBACK
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+SELECT * FROM animals WHERE species = 'digimon' OR species = 'pokemon'; 
+COMMIT;
+SELECT * FROM animals WHERE species = 'digimon' OR species = 'pokemon';
+
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals; -- Verify that all records were deleted
+ROLLBACK;
+SELECT * FROM animals;--After ROLLBACK
+
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT animals_born_after_Jan1st22;
+SELECT * FROM animals;
+UPDATE animals SET weight_kg = weight_kg * -1;
+SELECT * FROM animals;
+ROLLBACK TO animals_born_after_Jan1st22;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+SELECT * FROM animals;
+
+SELECT COUNT(*) FROM animals;--Animals count
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;--How many animals have never tried to escape?
+SELECT AVG(weight_kg) FROM animals;-- the average weight of animals
+
+SELECT neutered, SUM(escape_attempts) AS total_attempts
+FROM animals
+GROUP BY neutered
+ORDER BY total_attempts DESC
+LIMIT 1;--Who escapes the most, neutered or not neutered animals?
```

Comment:
- [ ] It would be better to fix this part of the query, we are expecting who escapes the most, neutered or not neutered. Instead of using `SUM`, let us use `AVG` or `MAX` in order to achieve the correct query results. Kindly remember to update the screenshot as well